python/src/bindings/binding_helpers.hpp

```diff
@@ -117,16 +117,24 @@ namespace phylanx { namespace bindings
 
     ///////////////////////////////////////////////////////////////////////////
     inline phylanx::execution_tree::primitive_argument_type
-    expression_compiler(std::string xexpr, pybind11::args args)
+    expression_compiler(std::string xexpr_str, pybind11::args args)
     {
         namespace et = phylanx::execution_tree;
         return hpx::threads::run_as_hpx_thread(
             [&]() -> et::primitive_argument_type
             {
                 try
                 {
-                    et::compiler::function_list eval_snippets;
-                    auto x = et::compile(xexpr, eval_snippets);
+                    static phylanx::execution_tree::compiler::environment *eval_env = nullptr;
```

Comment:
Why pointers? Where is the allocated memory free'd?

Reply:
It is not freed, of course. The plan was for a single, long-running set of defs to be available globally. My experience is that static objects don't initialize properly, so I use pointers to objects. I also tried making an extern version of these two objects, but this also caused memory problems.

Reply:
Static objects initialize just fine, starting with C++11 their initialization is thread-safe even (which your solution is not, btw).